lib/open_api_spex/controller.ex

```diff
@@ -0,0 +1,153 @@
+defmodule OpenApiSpex.Controller do
+  @moduledoc ~S'''
+  Generation of OpenAPI documentation via ExDoc documentation and tags.
+
+  ## Supported OpenAPI fields
+
+  Attribute `operationId` is automatically provided by the implementation
+  and cannot be changed in any way. It is constructed as `Module.Name.function_name`
+  in the same way as function references in backtraces.
+
+  ### `description` and `summary`
+
+  Description of endpoint will be filled with documentation string in the same
+  manner as ExDocs, so first line will be used as a `summary` and whole
+  documentation will be used as `description` field.
+
+  ### `parameters`
+
+  Parameters of the endpoint are defined by `:parameters` tag which should be
+  map or keyword list that is formed as:
+
+  ```
+  [
+    param_name: definition
+  ]
+  ```
+
+  Where `definition` is `OpenApiSpex.Parameter.t()` structure or map or keyword
+  list that accepts the same arguments.
+
+  ### `responses`
+
+  Responses are controlled by `:responses` tag and currently supports only
+  `application/json` responses. Responses must be defined as a map or keyword list
+  in form of:
+
+  ```
+  %{
+    200 => {"Response name", schema},
+    :not_found => {"Response name", schema}
+  }
+  ```
+
+  Where atoms are the same as `Plug.Conn.Status.code/1` values.
+
+  ### `requestBody`
+
+  Controlled by `:body` parameter and is defined as a tuple in form
+  `{description, schema}`.
+
+  ### `tags`
+
+  Tags are controlled by `:tags` attribute. In contrast to other attributes, this
+  one will also inherit all tags defined as a module documentation attributes.
+
+  ## Example
+
+  ```
+  defmodule FooController do
+    use #{inspect(__MODULE__)}
+
+    @moduledoc tags: ["Foos"]
+
+    @doc """
+    Endpoint summary
+
+    More docs
+    """
+    @doc parameters: [
+      id: [in: :path, type: :string, required: true]
+    ]
+    @doc responses: [
+      ok: {"Foo document", FooSchema}
```

Comment:
OpenAPI allows a different response schema for different content types. This locks the user into one content type. I don't think this interface is flexible enough.

Reply:
@moxley it will lock only if you use maps, `[foo: 1, foo: 2]` is perfectly valid in Elixir, however it will not work with current [API](https://hexdocs.pm/open_api_spex/OpenApiSpex.Responses.html#t:t/0) which on the other hand truly restricts one content type per response.

Reply:
So you'd solve this by allowing a separate `@doc responses: [...]` for each content type?

Reply:
No, it is perfectly possible with keyword list to define:

```
@doc responses: [
  ok: {"foo", "application/foo", schema},
  ok: {"bar", "application/bar", schema}
]
```

With current `attributes` syntax, however it will not work on the `open_api_spex` as `responses` field expects map that naturally allow only one value per key. So this isn't limitation of this syntax, but limitation of the library.